src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -53,6 +53,13 @@ public MemoryCache(IOptions<MemoryCacheOptions> optionsAccessor!!, ILoggerFactor
                 _options.Clock = new SystemClock();
             }
 
+            if (_options.TrackStatistics)
+            {
+                _allStats = new();
+                _accumulatedStats = new Stats(this, skipAdd: true);
+                _stats = new ThreadLocal<Stats>(() => new Stats(this));
```

Comment:
What happens if someone sets TrackStatistics to true after the cache is already created? The code appears to be checking the live value on every operation, but these fields may not have been initialized., right?

Reply:
Should the other `if (_options.TrackStatistics)` be changed to `_allStats is not null` or something? I don't think that we should support changing `TrackStatistics` to true after the cache is already created. That's not really the intent of the option being settable. It's that way to support the "options pattern."

Reply:
Switched the flag. TrackStatistics is not expected to change for the lifetime of the cache.